Mathlib/Algebra/Order/Ring/Defs.lean

```diff
@@ -141,7 +141,10 @@ class OrderedSemiring (α : Type u) extends Semiring α, OrderedAddCommMonoid α
 
 /-- An `OrderedCommSemiring` is a commutative semiring with a partial order such that addition is
 monotone and multiplication by a nonnegative number is monotone. -/
-class OrderedCommSemiring (α : Type u) extends OrderedSemiring α, CommSemiring α
+class OrderedCommSemiring (α : Type u) extends OrderedSemiring α, CommSemiring α where
+  mul_le_mul_of_nonneg_right a b c ha hc :=
+    -- parentheses ensure this generates an `optParam` rather than an `autoParam`
+    (by simpa only [mul_comm] using mul_le_mul_of_nonneg_left a b c ha hc)
```

Comment:
@digama0, can you confirm I'm not imagining that `()` makes a difference here? Is my explanation correct?

I could of course just write out the term-mode proof, but this seemed clearer.

Reply:
I don't think auto params work in this position. That sounds like a feature request (after which you would have to use this paren trick here).

Reply:
How do we make this feature request?